syncer/syncer.go

```diff
@@ -2117,7 +2127,9 @@ func (s *Syncer) Close() {
 // stopSync stops syncing, now it used by Close and Pause
 // maybe we can refine the workflow more clear
 func (s *Syncer) stopSync() {
-	<-s.done // wait Run to return
+	if s.done != nil {
```

Comment:
Is it better to add this check to `close(s.done)` in function `func (s *Syncer) Run(ctx context.Context) (err error)`, although we can ensure `s.Process` is always called before `s.Run` and `s.done` is initialized in `s.Process`

Reply:
yes, it's better, I fixed it.